packages/workbox-broadcast-cache-update/BroadcastCacheUpdatePlugin.mjs

```diff
@@ -0,0 +1,64 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import {assert} from 'workbox-core/_private/assert.mjs';
+import {BroadcastCacheUpdate} from './BroadcastCacheUpdate.mjs';
+import './_version.mjs';
+
+/**
+ * This plugin will automatically broadcast a message whenever a cached response
+ * is updated.
+ *
+ * @memberof workbox.broadcastUpdate
+ */
+class BroadcastCacheUpdatePlugin extends BroadcastCacheUpdate {
```

Comment:
Could you match what I ended up doing with `CacheableResponsePlugin`: don't extend the base class, but instead take in identical constructor parameters and create an internal instance that you then use in `cacheDidUpdate`?

https://github.com/GoogleChrome/workbox/pull/1057/files#diff-7ecb07c7bc862719cacd71553537e137R39

Reply:
Is there a reason for this preference?